app/services/mixtape_maker.rb

```diff
@@ -0,0 +1,15 @@
+class MixtapeMaker
+
+  def self.generate(no_song = 5, artists:)
+    new.generate(no_song, artists: artists)
+  end
+
+  def generate(no_song = 5, artists:)
+    if artists.blank?
+      Song.limit(no_song).order("RANDOM()")
+    else
+        Song.where(artist: artists[:id]).limit(no_song).order("RANDOM()")
```

Comment:
Tend to put the `limit` last ... since that's what SQL will do.

Plus, it's like read, "I want songs for this artist in random order, but just 5 of them"

Ok, that isn't a great example, but LIMIT is the last part of the SQL query) so it is nice to match its structure.